### PR TITLE
minor fixes in deploy button

### PR DESCRIPTION
**Pull request by sub-mod**
This PR fixes the issue when no error is thrown when deploy button is pressed while apiserver is not up
With the fix, we check if we can successfully GET from apiserver and if so create all entities

### CODE REVIEW

File: pkg/kubernetes/app.js [hunk 1]
```diff
@@ -30,6 +30,16 @@ define([
             calculated.containers = "0";
 
             /*
+             * Calculate total number of containers = replicas * containersArray.length
+             */
+            client.select(service.spec.selector || { },
+                          service.metadata.namespace, "ReplicationController").forEach(function(rc) {
```
Comment (stefwalter) [inline pkg/kubernetes/app.js]: This is an interesting approach. But I don't this actually works.

In particular, both services and reciplicationcontrollers select pods. But services do not select replicationcontrololers, or vice versa.

Comment (stefwalter) [inline pkg/kubernetes/app.js]: I would suggest sticking with the previous approach where we enumerate the pods. Each pod has a spec which says how many containers are expected in that pod. We would just use that number.

File: pkg/kubernetes/deploy.js [hunk 2]
```diff
@@ -105,39 +103,55 @@ define([
...
-                })
-                .done(function() {
-                    /* code gets run when everything is created */
-                    disable_deploy_button();
-                    hide_progress_message();
-                     $('#deploy-app-dialog').modal('hide');
+            show_progress_message();
+            deploy_btn.prop("disabled", true);
 
+            client.get_services()
```
Comment (stefwalter) [inline pkg/kubernetes/deploy.js]: Why are we doing this?

**Comment by stefwalter** [comment]
In any case the two parts here should be split out into two or more patches that do distinct things with commit messages that describe the changes.

File: pkg/kubernetes/deploy.js (hunk 2 above)
Comment (sub-mod) [inline pkg/kubernetes/deploy.js]: I need to check if kubernetes daemons are up and the apiserver is functioning well by calling a REST api.
and if if it works create all entities.

Comment (stefwalter) [inline pkg/kubernetes/deploy.js]: I'll be working on a curtain implementation, as was highlighted by Jeff, and the Openshift work. Meanwhile, we can assume that apiserver is up if the dialog is visible.

**Comment by stefwalter** [comment]
I don't see a description of what is being fixed in either the commit message or this pull request.

**Comment by stefwalter** [comment]
This wasn't cleaned up. so I included it in another commit that now replaces it: #2271

**Pull request closed without merging**